packages/common-all/src/error.ts

```diff
@@ -45,15 +44,70 @@ type ServerErrorProps = {
   code?: StatusCodes;
 };
 
-export type IDendronError = DendronErrorProps;
+export type IDendronError<TCode = StatusCodes | undefined> =
+  DendronErrorProps<TCode>;
 
-export class DendronError extends Error implements IDendronError {
+export abstract class BaseDendronError<TCode = StatusCodes | undefined>
+  extends Error
+  implements IDendronError<TCode>
+{
   public status?: string;
   public payload?: string;
   public severity?: ERROR_SEVERITY;
-  public code?: number;
+  public code?: TCode;
   public innerError?: Error;
 
+  /** The output that may be displayed to a person if they encounter this error. */
+  public stringifyForHumanReading() {
+    return this.message;
+  }
+
+  /** Overload this to change how the `payload` is stringified. */
+  protected payloadStringify() {
+    return JSON.stringify(this.payload);
+  }
+
+  /** The output that may be saved into the local logs for the user. */
+  public stringifyForLogs() {
+    const { severity, code, message } = this;
+    const payload: { [key: string]: any } = {
+      severity,
+      code,
+      message,
+    };
+    if (this.innerError) {
+      payload.innerError = this.innerError;
+    }
+    if (this.payload) {
+      payload.payload = this.payloadStringify();
+    }
+    return JSON.stringify(payload);
+  }
+
+  /** The output that may be sent to Sentry, or other telemetry service.
+   *
+   * This function will eventually check that the output is stripped of PII,
+   * but for now that's the same as these.
+   */
+  public stringifyForTelemetry() {
+    return this.stringifyForLogs();
+  }
+
+  /** Returns the HTTP error code equivalent for this error.
+   *
+   * Override this to change
+   */
+  abstract translateToHTTPErrorCode(): StatusCodes | undefined;
+
+  /** If false, this error does not necessarily mean the operation failed. It should be possible to recover and resume. */
+  public isFatal() {
```

Comment:
nit: can we use getter syntactic sugar? 

public get isFatal(): boolean {
  ...
}